applications/serial_lte_modem/doc/slm_extending.rst

```diff
@@ -7,46 +7,33 @@ Extending the application
    :local:
    :depth: 2
 
-You can extend the serial LTE modem application by adding your own AT commands and custom error handling.
+The AT commands in the Serial LTE modem are implemented with the :ref:`at_cmd_custom_readme` library.
+You can extend the Serial LTE modem application by adding your own AT commands.
 
 Adding AT commands
 ******************
 
-If you want to implement custom AT commands and add them to the serial LTE modem application, see the parser implementation of the provided proprietary AT commands for reference.
-The source files can be found in the :file:`applications/serial_lte_modem/src/` folder.
+If you want to implement a custom AT command in SLM, you will need to follow the instructions in the :ref:`at_cmd_custom_readme`.
+Setting the AT commands filter and your callback with the :c:macro:`AT_CMD_CUSTOM` macro is enough for a custom AT command that does not require setup or teardown.
 
-Complete the following steps to add a parser for your own AT commands:
+If your custom AT command requires setup or teardown, you will need to perform the following steps:
 
 1. Create a header file in :file:`applications/serial_lte_modem/src/`.
-   The file must expose the following functions:
+   The file must expose the following function declarations:
 
-   * ``*_init()`` - Initialize the parser.
-   * ``*_uninit()`` - Uninitialize the parser.
+   * ``*_init()`` - Your setup function for the custom AT command.
+   * ``*_uninit()`` - Your teardown function for the custom AT command.
 
-   See the files for existing AT command parsers for reference.
-#. Implement your AT strings and handlers in a corresponding :file:`.c` file.
-   See the files for existing AT command parsers for reference.
+#. Create a corresponding :file:`.c` file in :file:`applications/serial_lte_modem/src/`.
+   The file must implement the following functions:
 
-   Pay attention to the following requirements:
+   * Your AT command callback, which is filtered with the :c:macro:`AT_CMD_CUSTOM` macro.
+   * ``*_init()`` - Your setup function for the custom AT command.
+   * ``*_uninit()`` - Your teardown function for the custom AT command.
+     Make sure that the uninit function exits successfully. It will be called before entering power saving states or shutting down.
```

Comment:
```suggestion
     Make sure that the teardown function exits successfully.
     It will be called before entering power saving states or shutting down.
```